packages/notifications/src/pushNotifications/providers/pinpoint/apis/requestPermissions.ts

```diff
@@ -0,0 +1,38 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { PlatformNotSupportedError } from '@aws-amplify/core/internals/utils';
+import { RequestPermissions } from '../types';
+
+/**
+ * Requests notification permissions from your user. By default, Amplify requests all supported permissions but you can
+ * choose not to request specific permissions. The resulting promise will resolve to true if requested permissions are
+ * granted (or have previously been granted) or false otherwise. Not all specific permissions are supported by platforms
+ * your React Native app can run on but will be safely ignored even on those plaforms. Currently supported permissions:
+ *
+ *   Alert: When set to true, requests the ability to display notifications to the user.
+ *
+ *   Sound: When set to true, requests the ability to play a sound in response to notifications.
+ *
+ *   Badge: When set to true, requests the ability to update the app's badge.
+ *
```

Comment:
Should this inline doc have description of the parameters? (same to other APIs inline doc)

```suggestion
 *   * `alert`: When set to true, requests the ability to display notifications to the user.
 *
 *   * `sound`: When set to true, requests the ability to play a sound in response to notifications.
 *
 *   * `badge`: When set to true, requests the ability to update the app's badge.
 *
```